### PR TITLE
Adding metadata

### PR DESCRIPTION
**Pull request by carvilsi**
Removing old way to deal with metadata related with apps and providing a more general way to do this.
Related with:
CMDB-96
WE-33
WE-34

I certify that:
- [x] The automated build passes on the branch this pull request is from
- [x] My change is documented in the Changelist.md
- [x] Any functionality additions/changes are documented in the README.md
- [x] Any new dockerimages this build depends on are properly tagged on github
- [x] I shall not merge until the travis build is green and the PR is accepted by
  another developer
- [x] I shall promptly fix/reply to comments, as I know PRs are not :fire: and
  forget
- [x] I shall keep my branch updated with any intermediate changes to the
  target branch
- [x] This is a quality PR
- [x] I got to the end of this list

### CODE REVIEW

File: test/WeaverProject.test.coffee [hunk 1]
```diff
@@ -73,26 +73,56 @@ describe 'WeaverProject Test', ->
...
+    key1 = 'model1'
+    bundleKey = 'models'
+    objectToTest = {
+      meta0
+      meta1
+    } 
+    Promise.join(p.addMetadata(bundleKey,key0,meta0),p.addMetadata(bundleKey,key1,meta1),->
+      p.getMetadata(bundleKey, null)
+    ).then((metadataFromBundle)->
+      assert(metadataFromBundle, objectToTest)
```
Comment (bastbijl) [inline test/WeaverProject.test.coffee]: use two `assert.deepEqual` for each added metadata bundle `meta0` and `meta1`

Comment (carvilsi) [reply test/WeaverProject.test.coffee]: 👍

Comment (gvdent) [reply test/WeaverProject.test.coffee]: (Either that or `expect(objectToTest).to.eql(metadataFromBundle)`)

File: src/CoreManager.coffee [hunk 2]
```diff
@@ -216,11 +216,14 @@ class CoreManager
   unfreezeProject: (id) ->
     @GET("project.unfreeze", {id}, id)
 
-  addApp: (id, appName, appMetadata) ->
```
[the remark this answers is not in the document]
Comment (carvilsi) [reply src/CoreManager.coffee]: I do not see here the needs for backward compatibility, take in account that we are restricting the required weaver-sdk Vs weaver-server, connector... And in the other hand:

`The backwards compatibility is already take in account, only weaver-electron was using this and even there was never used on production. In the other hand this is already update in another PR for weaver-electron.`

But maybe I'm wrong. What do you think @gvdent?

[the remark this answers is not in the document]
Comment (gvdent) [reply src/CoreManager.coffee]: My opinion (slight rant warning) is posted here: https://github.com/weaverplatform/weaver-server/pull/168#issuecomment-389836275

**Pull request merged**